src/GraphQL.Tests/Utilities/GetGraphTypeFromTypeTests.cs

```diff
@@ -223,11 +223,145 @@ public void GetGraphTypeFromType_Matrix(Type type, bool nullable, TypeMappingMod
+            schema.Initialize();
+            schema.Query.Fields.Find("test").Arguments.Find("arg").Type.ShouldBe(mappedType);
+        }
+
+        [Theory]
+        [InlineData(typeof(IntGraphType), typeof(IntGraphType))]
+        [InlineData(typeof(StringGraphType), typeof(StringGraphType))]
+        [InlineData(typeof(ListGraphType<StringGraphType>), typeof(ListGraphType<StringGraphType>))]
+        [InlineData(typeof(NonNullGraphType<StringGraphType>), typeof(NonNullGraphType<StringGraphType>))]
+        [InlineData(typeof(MyClassInputType), typeof(MyClassInputType))]
+        [InlineData(typeof(NonNullGraphType<ListGraphType<ListGraphType<NonNullGraphType<MyClassInputType>>>>), typeof(NonNullGraphType<ListGraphType<ListGraphType<NonNullGraphType<MyClassInputType>>>>))]
+        [InlineData(typeof(GraphQLClrInputTypeReference<int>), typeof(IntGraphType))]
+        [InlineData(typeof(GraphQLClrInputTypeReference<string>), typeof(StringGraphType))]
+        [InlineData(typeof(ListGraphType<GraphQLClrInputTypeReference<string>>), typeof(ListGraphType<StringGraphType>))]
+        [InlineData(typeof(NonNullGraphType<GraphQLClrInputTypeReference<string>>), typeof(NonNullGraphType<StringGraphType>))]
+        [InlineData(typeof(GraphQLClrInputTypeReference<MyClass>), typeof(MyClassInputType))]
+        [InlineData(typeof(GraphQLClrInputTypeReference<MyEnum>), typeof(EnumerationGraphType<MyEnum>))]
+        [InlineData(typeof(GraphQLClrInputTypeReference<MappedEnum>), typeof(MappedEnumGraphType))]
+        [InlineData(typeof(NonNullGraphType<ListGraphType<ListGraphType<NonNullGraphType<GraphQLClrInputTypeReference<MyClass>>>>>), typeof(NonNullGraphType<ListGraphType<ListGraphType<NonNullGraphType<MyClassInputType>>>>))]
+        public void InputTypeIsDereferenced_InputField(Type referenceType, Type mappedType)
+        {
+            var inputType = new InputObjectGraphType();
+            inputType.Field(referenceType, "field");
+            var query = new ObjectGraphType();
+            query.Field(typeof(IntGraphType), "test",
+                arguments: new QueryArguments
+                {
+                    new QueryArgument(inputType) { Name = "arg" }
+                });
+            var schema = new Schema()
+            {
+                Query = query
+            };
+            schema.RegisterTypeMapping(typeof(MyClass), typeof(MyClassObjectType));
+            schema.RegisterTypeMapping(typeof(MyClass), typeof(MyClassInputType));
+            schema.RegisterTypeMapping(typeof(MappedEnum), typeof(MappedEnumGraphType));
+            schema.Initialize();
+            var inputTypeActual = schema.Query.Fields.Find("test").Arguments.Find("arg").ResolvedType.ShouldBeOfType<InputObjectGraphType>();
+            inputTypeActual.ShouldBe(inputType);
+            inputTypeActual.Fields.Find("field").Type.ShouldBe(mappedType);
+        }
+
+        private class MyPairObjectType : ObjectGraphType<KeyValuePair<int, string>>
```

Comment:
Is not used.

Reply:
Correct. Old code from first pass at tests. I’ll delete it tonight.